domain/network/state/subnet.go

```diff
@@ -348,22 +353,25 @@ func (st *State) GetSubnetsByCIDR(
 	q := `
 SELECT &SubnetRow.*
 FROM   v_space_subnet
-WHERE  subnet_cidr = $M.cidr`
+WHERE  subnet_cidr = $SubnetRow.subnet_cidr`
```

Comment:
How is this going to be used in practice? Specifically I think the service needs to set the state up for success a little more. Because `subnet_cidr` is a string we have no idea both how the subnet has been stored or how the caller has provided the subnet.

For example `2001:0DB8:0:0:a:0:0:0`. Is it `2001:0DB8::a:0:0:0` or `2001:0DB8:0:0:a::`. I think the rules dictate that the largest set of zero's should be replaced. In any event my point is that representation logic store for UI is not a good lookup mechanism. They should be parsed and converted to int range to be searched on in the DB layer to avoid subtle errors IMO.

Reply:
ok